### PR TITLE
Remove concept of types, supportedTypes, BasicCardType enum

### LINKED ISSUE #59: Drop supportedTypes?
As part of discussing PR API issue 753 [1] it has emerged that there are challenges in accurately determining at low cost the card type (credit, debit, prepaid). 

After asking a few of the WG participants whether they have heard significant demand for this feature, the initial responses have been "no." 

Therefore, I would like the group to explicitly consider whether to drop supportedTypes in this version of Basic Card.

[1] https://github.com/w3c/payment-request/issues/753

### PR DESCRIPTION
**Pull request by marcoscaceres**
closes #59 

The following tasks have been completed:

 * [x] Confirmed there are no ReSpec errors/warnings.
 * [x] Modified Web platform tests: [Payment Request](https://github.com/web-platform-tests/wpt/pull/13858), [Basic-Card](https://github.com/web-platform-tests/wpt/pull/13859), [Payment Handler](https://github.com/web-platform-tests/wpt/pull/13860)
 * [ ] added MDN Docs (link)

Implementation commitments:

 * [ ] Safari (link to issue)
 * [ ] Chrome (link to issue)
 * [x] [Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1504032)
 * [ ] Edge (public signal)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-method-basic-card/pull/62.html" title="Last updated on Nov 15, 2018, 5:44 AM GMT (1400274)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-method-basic-card/62/3c67354...1400274.html" title="Last updated on Nov 15, 2018, 5:44 AM GMT (1400274)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -57,28 +57,16 @@
...
     </section>
     <section id='sotd'>
       <p>
-        The working group maintains <a href=
-        "https://github.com/w3c/payment-method-basic-card/issues?utf8=%E2%9C%93&amp;q=is%3Aopen%20is%3Aissue%20-label%3A%22Priority%3A%20Postponed%22%20">
-        a list of all bug reports that the group has not yet addressed</a>.
-        Pull requests with proposed specification text for outstanding issues
-        are strongly encouraged.
+        This is a work in progress and is expected to change as the
+        specification gets implemented and deployed.
```
Comment (ianbjacobs) [inline index.html]: Editorial suggestion:
 "is expected to change with implementation and deployment experience."

**Review by ianbjacobs — APPROVED** [review]
Only one editorial suggestion; otherwise good to go.

**Comment by marcoscaceres** [comment]
@rsolomakhin you probably saw, I sent a bunch of pull requests updating the tests. I hacked up your handler tests, so please make sure I didn't screw them up :)   

@zouhir, heads up on the Microsoft side.

**Comment by rsolomakhin** [comment]
ACK

**Comment by zouhir** [comment]
👍

**Comment by marcoscaceres** [comment]
Oki, sent removal patch on the Gecko side...

**Comment by ianbjacobs** [comment]
Call for consensus to remove sent today:
 https://lists.w3.org/Archives/Public/public-payments-wg/2018Nov/0008.html

**Comment by ianbjacobs** [comment]
@marcoscaceres,

I believe there are libraries that support Basic Card. We want to understand the impact of deprecating this feature on that deployed software. Perhaps the editors can reach consensus on appropriate browser behavior when they encounter this deprecated feature.

Thanks!

**Comment by marcoscaceres** [comment]
Libraries would be unaffected in terms of behavior - passing in `supportedTypes` is ignored. And the spec makes no guarantees that the returned card is of the requested network or type, so nothing really changes there: the payment processor still needs to check that the user didn’t provide a bogus card.

**Comment by marcoscaceres** [comment]
Blocked on CFC until 15th of November.

**Pull request merged**